framework/src/modules/dpos_v2/module.ts

```diff
@@ -132,10 +147,273 @@ export class DPoSModule extends BaseModule {
 		voteCommand.init({ tokenIDDPoS: this._moduleConfig.tokenIDDPoS });
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async afterGenesisBlockExecute(_context: GenesisBlockExecuteContext): Promise<void> {
-		// eslint-disable-next-line no-console
-		console.log(this._bftAPI, this._randomAPI, this._validatorsAPI, this._moduleConfig);
+	public async afterGenesisBlockExecute(context: GenesisBlockExecuteContext): Promise<void> {
+		const assetBytes = context.assets.getAsset(this.id);
+		// if there is no asset, do not initialize
+		if (!assetBytes) {
+			return;
+		}
+		const genesisStore = codec.decode<GenesisStore>(genesisStoreSchema, assetBytes);
+		const errors = validator.validate(genesisStoreSchema, genesisStore);
+
+		if (errors.length > 0) {
+			throw new LiskValidationError(errors);
+		}
+
+		// validators property check
+		const dposValidatorAddresses = [];
+		const dposValidatorNames = [];
+		const dposValidatorAddressMap = new dataStructures.BufferMap();
+		for (const dposValidator of genesisStore.validators) {
+			if (!isUsername(dposValidator.name)) {
+				throw new Error(`Invalid validator name ${dposValidator.name}.`);
+			}
+			dposValidatorAddressMap.set(dposValidator.address, true);
+			dposValidatorAddresses.push(dposValidator.address);
+			dposValidatorNames.push(dposValidator.name);
+		}
+		if (!objectUtils.bufferArrayUniqueItems(dposValidatorAddresses)) {
+			throw new Error('Validator address is not unique.');
+		}
+		if (new Set(dposValidatorNames).size !== dposValidatorNames.length) {
+			throw new Error('Validator name is not unique.');
+		}
+		// voters property check
+		const voterAddresses = [];
+		for (const voter of genesisStore.voters) {
+			if (voter.sentVotes.length > MAX_VOTE) {
+				throw new Error(`Sent vote exceeds max vote ${MAX_VOTE}.`);
+			}
+			if (!objectUtils.bufferArrayUniqueItems(voter.sentVotes.map(v => v.delegateAddress))) {
+				throw new Error('Sent vote delegate address is not unique.');
+			}
+			if (!objectUtils.bufferArrayOrderByLex(voter.sentVotes.map(v => v.delegateAddress))) {
+				throw new Error('Sent vote delegate address is not lexicographically ordered.');
+			}
+			if (voter.sentVotes.some(v => !dposValidatorAddressMap.has(v.delegateAddress))) {
+				throw new Error('Sent vote includes non existing validator address.');
+			}
+			if (voter.pendingUnlocks.length > MAX_UNLOCKING) {
+				throw new Error(`PendingUnlocks exceeds max unlocking ${MAX_UNLOCKING}.`);
+			}
+			const sortingPendingUnlocks = [...voter.pendingUnlocks];
+			sortUnlocking(sortingPendingUnlocks);
+			for (let i = 0; i < voter.pendingUnlocks.length; i += 1) {
+				const original = voter.pendingUnlocks[i];
+				const target = sortingPendingUnlocks[i];
+				if (!equalUnlocking(original, target)) {
+					throw new Error('PendingUnlocks are not lexicographically ordered.');
+				}
```

Comment:
I think destructure an object is also not copying the `Buffer`, so it will still be confusing.
Also, here intention is to just sort and check if the original array was sorted or not, so i think it should be ok